UPGRADE.md

```diff
@@ -1413,6 +1413,49 @@ const PostShow = () => (
 };
 ```
 
+## No More Props Injection In `<AppBar>`
+
+When a React element was provided as the `userMenu` prop, the `<AppBar>` used to clone it and inject the `logout` prop (a React element). This is no longer the case and if you provided a custom user menu, you now have to include the logout yourself.
+
+```diff
+-import { UserMenu } from 'react-admin';
++import { Logout, UserMenu } from 'react-admin';
+
+-export const MyUserMenu = (props) => (
++export const MyUserMenu = () => (
+-    <UserMenu {...props}>
++    <UserMenu logout={<Logout />}>
+        <MenuItemLink
+            to="/configuration"
+            primaryText="pos.configuration"
+            leftIcon={<SettingsIcon />}
+            sidebarIsOpen
+        />
+    </UserMenu>
+)
+```
+
+## `<MenuItemLink>` Automatically Translates `primaryText`
+
+You can pass a translation key directly as the `primaryText` prop for `<MenuItemLink>`.
+
+```diff
+const CustomUserMenu = (props: any) => {
+-    const translate = useTranslate();
+    return (
+        <UserMenu {...props}>
+            <MenuItemLink
+                to="/configuration"
+-                primaryText={translate('pos.configuration')}
```

Comment:
what's the rationale behind this change? It doesn't save any cloneElement.

Reply:
Nothing, we just avoid having to get `useTranslate` and call `translate` ourselves